File: generate-icon.js
const { createCanvas, loadImage } = require('canvas');
const fs = require('fs');
const path = require('path');

function drawIcon(ctx, size) {
    const center = size / 2;
    const scale = size / 512;

    ctx.clearRect(0, 0, size, size);

    ctx.save();
    
    const radius = 70 * scale;
    ctx.beginPath();
    ctx.moveTo(radius, 0);
    ctx.arcTo(size, 0, size, size, radius);
    ctx.arcTo(size, size, 0, size, radius);
    ctx.arcTo(0, size, 0, 0, radius);
    ctx.arcTo(0, 0, size, 0, radius);
    ctx.closePath();
    
    const gradient = ctx.createLinearGradient(0, 0, size, size);
    gradient.addColorStop(0, '#667eea');
    gradient.addColorStop(1, '#764ba2');
    ctx.fillStyle = gradient;
    ctx.fill();

    ctx.shadowColor = 'rgba(0,0,0,0.35)';
    ctx.shadowBlur = 20 * scale;
    ctx.shadowOffsetX = 0;
    ctx.shadowOffsetY = 5 * scale;

    const docX = 120 * scale;
    const docY = 110 * scale;
    const docW = 160 * scale;
    const docH = 210 * scale;
    
    ctx.fillStyle = '#ffffff';
    ctx.shadowColor = 'rgba(0,0,0,0.15)';
    ctx.shadowBlur = 20 * scale;
    ctx.shadowOffsetX = 0;
    ctx.shadowOffsetY = 6 * scale;
    roundRect(ctx, docX, docY, docW, docH, 8 * scale);
    ctx.fill();
    ctx.shadowBlur = 0;

    ctx.strokeStyle = 'rgba(102, 126, 234, 0.3)';
    ctx.lineWidth = 3 * scale;
    for (let i = 0; i < 4; i++) {
        ctx.beginPath();
        ctx.moveTo(docX + 20 * scale, docY + 45 * scale + i * 35 * scale);
        ctx.lineTo(docX + docW - 20 * scale, docY + 45 * scale + i * 35 * scale);
        ctx.stroke();
    }

    ctx.fillStyle = 'rgba(102, 126, 234, 0.8)';
    ctx.beginPath();
    ctx.moveTo(docX + docW, docY);
    ctx.lineTo(docX + docW - 40 * scale, docY);
    ctx.lineTo(docX + docW, docY + 40 * scale);
    ctx.closePath();
    ctx.fill();

    const sheetX = 240 * scale;
    const sheetY = 170 * scale;
    const sheetW = 150 * scale;
    const sheetH = 180 * scale;

    ctx.fillStyle = '#f0f4ff';
    ctx.shadowColor = 'rgba(0,0,0,0.2)';
    ctx.shadowBlur = 25 * scale;
    ctx.shadowOffsetX = 3 * scale;
    ctx.shadowOffsetY = 8 * scale;
    roundRect(ctx, sheetX, sheetY, sheetW, sheetH, 8 * scale);
    ctx.fill();
    ctx.shadowBlur = 0;

    ctx.strokeStyle = 'rgba(102, 126, 234, 0.25)';
    ctx.lineWidth = 2 * scale;
    
    for (let i = 0; i < 5; i++) {
        ctx.beginPath();
        ctx.moveTo(sheetX, sheetY + 30 * scale + i * 30 * scale);
        ctx.lineTo(sheetX + sheetW, sheetY + 30 * scale + i * 30 * scale);
        ctx.stroke();
    }
    
    ctx.beginPath();
    ctx.moveTo(sheetX + 45 * scale, sheetY);
    ctx.lineTo(sheetX + 45 * scale, sheetY + sheetH);
    ctx.moveTo(sheetX + 90 * scale, sheetY);
    ctx.lineTo(sheetX + 90 * scale, sheetY + sheetH);
    ctx.stroke();

    ctx.fillStyle = '#667eea';
    roundRect(ctx, sheetX + 8 * scale, sheetY + 8 * scale, 30 * scale, 16 * scale, 4 * scale);
    ctx.fill();
    ctx.fillStyle = '#764ba2';
    roundRect(ctx, sheetX + 55 * scale, sheetY + 8 * scale, 30 * scale, 16 * scale, 4 * scale);
    ctx.fill();

    ctx.restore();
}

function roundRect(ctx, x, y, width, height, radius) {
    ctx.beginPath();
    ctx.moveTo(x + radius, y);
    ctx.lineTo(x + width - radius, y);
    ctx.quadraticCurveTo(x + width, y, x + width, y + radius);
    ctx.lineTo(x + width, y + height - radius);
    ctx.quadraticCurveTo(x + width, y + height, x + width - radius, y + height);
    ctx.lineTo(x + radius, y + height);
    ctx.quadraticCurveTo(x, y + height, x, y + height - radius);
    ctx.lineTo(x, y + radius);
    ctx.quadraticCurveTo(x, y, x + radius, y);
    ctx.closePath();
}

async function main() {
    const iconsDir = path.join(__dirname, 'src-tauri', 'icons');
    
    if (!fs.existsSync(iconsDir)) {
        fs.mkdirSync(iconsDir, { recursive: true });
    }

    const sizes = [512, 256, 128, 64, 32, 16];

    for (const size of sizes) {
        const canvas = createCanvas(size, size);
        const ctx = canvas.getContext('2d');
        drawIcon(ctx, size);
        
        const buffer = canvas.toBuffer('image/png');
        
        if (size === 512) {
            fs.writeFileSync(path.join(iconsDir, 'icon.png'), buffer);
        }
        fs.writeFileSync(path.join(iconsDir, `${size}x${size}.png`), buffer);
        
        if ([32, 64, 128, 256, 512].includes(size)) {
            fs.writeFileSync(path.join(iconsDir, `${size}x${size}@2x.png`), buffer);
        }
    }

    console.log(`Icon generation complete! Icons saved to ${iconsDir}');
}

main().catch(console.error);

File: generate-icon.py
#!/usr/bin/env python3
from PIL import Image, ImageDraw, ImageFont, ImageFilter
import os

def generate_icon(size):
    img = Image.new('RGBA', (size, size), (0, 0, 0, 0))
    draw = ImageDraw.Draw(img)
    
    center = size // 2
    scale = size / 512
    
    radius = int(70 * scale)
    draw.rounded_rectangle([0, 0, size, size], radius=radius, fill=None)
    
    gradient = Image.new('RGBA', (size, size))
    for y in range(size):
        for x in range(size):
            t = (x + y) / (2 * size)
            r1, g1, b1 = 102, 126, 234
            r2, g2, b2 = 118, 75, 162
            r = int(r1 + (r2 - r1) * t)
            g = int(g1 + (g2 - g1) * t)
            b = int(b1 + (b2 - b1) * t)
            gradient.putpixel((x, y), (r, g, b, 255))
    
    mask = Image.new('L', (size, size), 0)
    mask_draw = ImageDraw.Draw(mask)
    mask_draw.rounded_rectangle([0, 0, size, size], radius=radius, fill=255)
    
    img = Image.composite(gradient, img, mask)
    
    doc_x = int(120 * scale)
    doc_y = int(110 * scale)
    doc_w = int(160 * scale)
    doc_h = int(210 * scale)
    
    doc = Image.new('RGBA', (size, size), (0, 0, 0, 0))
    doc_draw = ImageDraw.Draw(doc)
    doc_draw.rounded_rectangle([doc_x, doc_y, doc_x + doc_w, doc_y + doc_h], 
                               radius=int(8 * scale), fill=(255, 255, 255, 255))
    
    for i in range(4):
        line_y = doc_y + int(45 * scale) + i * int(35 * scale)
        doc_draw.line([doc_x + int(20 * scale), line_y, 
                      doc_x + doc_w - int(20 * scale), line_y],
                     fill=(102, 126, 234, 77), width=int(3 * scale))
    
    fold_pts = [
        (doc_x + doc_w, doc_y),
        (doc_x + doc_w - int(40 * scale), doc_y),
        (doc_x + doc_w, doc_y + int(40 * scale))
    ]
    doc_draw.polygon(fold_pts, fill=(102, 126, 234, 204))
    
    sheet_x = int(240 * scale)
    sheet_y = int(170 * scale)
    sheet_w = int(150 * scale)
    sheet_h = int(180 * scale)
    
    sheet = Image.new('RGBA', (size, size), (0, 0, 0, 0))
    sheet_draw = ImageDraw.Draw(sheet)
    sheet_draw.rounded_rectangle([sheet_x, sheet_y, sheet_x + sheet_w, sheet_y + sheet_h],
                                 radius=int(8 * scale), fill=(240, 244, 255, 255))
    
    for i in range(5):
        line_y = sheet_y + int(30 * scale) + i * int(30 * scale)
        sheet_draw.line([sheet_x, line_y, sheet_x + sheet_w, line_y],
                       fill=(102, 126, 234, 64), width=int(2 * scale))
    
    sheet_draw.line([sheet_x + int(45 * scale), sheet_y, sheet_x + int(45 * scale), sheet_y + sheet_h],
                   fill=(102, 126, 234, 64), width=int(2 * scale))
    sheet_draw.line([sheet_x + int(90 * scale), sheet_y, sheet_x + int(90 * scale), sheet_y + sheet_h],
                   fill=(102, 126, 234, 64), width=int(2 * scale))
    
    sheet_draw.rounded_rectangle([sheet_x + int(8 * scale), sheet_y + int(8 * scale),
                                  sheet_x + int(38 * scale), sheet_y + int(24 * scale)],
                                 radius=int(4 * scale), fill=(102, 126, 234, 255))
    sheet_draw.rounded_rectangle([sheet_x + int(55 * scale), sheet_y + int(8 * scale),
                                  sheet_x + int(85 * scale), sheet_y + int(24 * scale)],
                                 radius=int(4 * scale), fill=(118, 75, 162, 255))
    
    shadow_layer = Image.new('RGBA', (size, size), (0, 0, 0, 0))
    shadow_layer.paste(doc, (0, 0))
    shadow_layer.paste(sheet, (0, 0))
    shadow_blur = shadow_layer.filter(ImageFilter.GaussianBlur(int(10 * scale)))
    
    shadow_shifted = Image.new('RGBA', (size, size), (0, 0, 0, 0))
    shadow_shifted.paste(shadow_blur, (int(3 * scale), int(8 * scale)))
    shadow_shifted.putalpha(100)
    
    img = Image.alpha_composite(img, shadow_shifted)
    img = Image.alpha_composite(img, doc)
    img = Image.alpha_composite(img, sheet)
    
    return img

def main():
    icons_dir = os.path.join(os.path.dirname(__file__), 'src-tauri', 'icons')
    os.makedirs(icons_dir, exist_ok=True)
    
    sizes = [512, 256, 128, 64, 32, 16]
    
    for size in sizes:
        icon = generate_icon(size)
        if size == 512:
            icon.save(os.path.join(icons_dir, 'icon.png'))
        icon.save(os.path.join(icons_dir, f'{size}x{size}.png'))
        if size in [32, 64, 128, 256, 512]:
            icon.save(os.path.join(icons_dir, f'{size}x{size}@2x.png'))
    
    print(f'Icon generation complete! Icons saved to {icons_dir}')

if __name__ == '__main__':
    main()